controlplane/pkg/nsm/healer.go

```diff
@@ -0,0 +1,318 @@
+package nsm
+
+import (
+	"context"
+	"fmt"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/nsm"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/registry"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/serviceregistry"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+type networkServiceHealer interface {
+	healDstDown(healID string, connection *model.ClientConnection) bool
+	healDataplaneDown(healID string, connection *model.ClientConnection) bool
+	healRemoteDataplaneDown(healID string, connection *model.ClientConnection) bool
+	healDstNmgrDown(healID string, connection *model.ClientConnection) bool
+}
+
+type healer struct {
```

Comment:
I would prefer nsmHealProcessor in naming and nsm_heal_processor.go

Reply:
Done : 5fdc2c1